### PR TITLE
add nemesis: scaling

### PR DESCRIPTION
**Pull request by ethercflow**
### What problem does this PR solve? 
add a new nemesis
### What is changed and how does it work?
Periodic scale up and down with tidy-operator
### Check List 

Tests 

 - No code

Code changes

 - Has Go code change

Side effects

Related changes

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: pkg/nemesis/scaling.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+}
+
+func (s scalingGenerator) Generate(nodes []clusterTypes.Node) []*core.NemesisOperation {
+	ops := make([]*core.NemesisOperation, 1)
+	ops[0] = &core.NemesisOperation{
+		Type:        core.Scaling,
+		Node:        &nodes[0],
+		InvokeArgs:  nil,
+		RecoverArgs: nil,
+		RunTime:     time.Hour * time.Duration(7*24),
```
Comment (mahjonp) [inline pkg/nemesis/scaling.go]: `RunTime` means duration between injection and recovery this time, not means what's your understanding.

File: pkg/nemesis/scaling.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+type originReplicaNum struct {
+	TiDB int32
+	PD   int32
+	TiKV int32
+}
+
+func (s scaling) Invoke(ctx context.Context, node *clusterTypes.Node, args ...interface{}) error {
+	log.Printf("Creating scaling with ns %s\n", node.Namespace)
+	var n int32 = -2
+	for {
```
Comment (mahjonp) [inline pkg/nemesis/scaling.go]: `Invoke` is a instant action.

File: pkg/nemesis/scaling.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+			}
+		case <-ctx.Done():
+			return nil
+		}
+	}
+}
+
+func (s scaling) Recover(ctx context.Context, node *clusterTypes.Node, args ...interface{}) error {
+	log.Printf("Recover scaling with ns %s\n", node.Namespace)
+	return s.scaleCluster(ctx, node.Namespace, node.Namespace, 0)
```
Comment (mahjonp) [inline pkg/nemesis/scaling.go]: splits `Invoke` and move recovery logic here.

**Comment by zhouqiang-cl** [comment]
Did this have already test and worked

**Pull request merged**